graphgen_test.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"math"
+	"strings"
+
+	"testing"
+
+	"github.com/arangodb/feed/pkg/datagen"
+	"github.com/arangodb/feed/pkg/graphgen"
+)
+
+func numberElementsInChannel(c chan *datagen.Doc) (uint64, chan *datagen.Doc) {
+	v := make(chan *datagen.Doc, 1000)
+	var numElements uint64 = 0
+	for element := range c {
+		numElements++
+		v <- element
+	}
+	close(v)
+	return numElements, v
+}
+
+func testExpectedNumberVerticesEdges(t *testing.T, gg graphgen.GraphGenerator,
+	expectedNumVertices uint64, expectedNumEdges uint64) {
+
+	actualNumVertices, _ := numberElementsInChannel(gg.VertexChannel())
+	obtainedNumVertices := gg.NumberVertices()
+
+	actualNumEdges, _ := numberElementsInChannel(gg.EdgeChannel())
+	obtainedNumEdges := gg.NumberEdges()
+
+	if actualNumVertices != expectedNumVertices {
+		t.Fatalf("Wrong number of vertices. Expected: %v, actual: %v",
+			expectedNumVertices, actualNumVertices)
+	}
+	if obtainedNumVertices != expectedNumVertices {
+		t.Fatalf("Wrong number of vertices. Expected: %v, obtained: %v",
+			expectedNumVertices, obtainedNumVertices)
+	}
+
+	if actualNumEdges != expectedNumEdges {
+		t.Fatalf("Wrong number of edges. Expected: %v, actual: %v",
+			expectedNumEdges, actualNumEdges)
+	}
+	if obtainedNumEdges != expectedNumEdges {
+		t.Fatalf("Wrong number of edges. Expected: %v, obtained: %v",
+			expectedNumEdges, obtainedNumEdges)
+	}
+}
+
+func TestCycleGeneration(t *testing.T) {
+	var length uint64 = 5
+	cycleGenerator, err := (&graphgen.CycleGraphParameters{length, ""}).MakeGraphGenerator()
```

Comment:
This compound literal produces a warning:
```graphgen_test.go|67-68 col 26-27 warning| [composites] github.com/arangodb/feed/pkg/graphgen.PathParameters composite literal uses unkeyed fields [W]```
I think we should use
```(&graphgen.CycleGraphParameters{Length: length, Prefix: ""})```
even though this is a bit more verbose. There are a few other places in this file with the same thing.

Reply:
I don't get any warnings, how do you produce them?

Reply:
Maybe different go versions. I am using 1.19.2. I just got it by the usual `go build`.